synapse/lib/cell.py

```diff
@@ -1679,6 +1679,8 @@ async def promote(self, graceful=False):
             mesg = 'promote() called on non-mirror'
             raise s_exc.BadConfValu(mesg=mesg)
 
+        logger.warning(f'PROMOTION: Performing leadership promotion {graceful=} ahaname={self.conf.get("aha:name")}')
```

Comment:
Recommend one of the following:
1. Move the `ahaname = self.conf.get('aha:name')` from L1686 to the top of the function and use `ahaname` in all of these log calls
2. Use `%s` in the logging format string so the argument gets lazily evaluation

Reply:
We use fstrings pretty consistently with our logs ( and fstrings are faster to eval than ``%`` strings ); as this isn't a tight loop I'm not concerned about string formatting having a performance penalty on string formatting operations.

Reply:
If this was perf oriented code i'd def want the lazy evaluation, but it's ok for this :+1: